libtenzir/src/node.cpp

```diff
@@ -354,6 +420,12 @@ node(node_actor::stateful_pointer<node_state> self, std::string /*name*/,
     // Core components are terminated in a second stage, we remove them from the
     // registry upfront and deal with them later.
     std::vector<caf::actor> core_shutdown_handles;
+    for (const auto& name :
+         std::ranges::reverse_view(self->state.ordered_components)) {
```

Comment:
Could be `self->state.ordered_components | std::ranges::views::reverse`, which I think is a bit more idiomatic.